src/pages/staticPages/tools/oncoprinter/OncoprinterClinicalAndHeatmapUtils.ts

```diff
@@ -0,0 +1,572 @@
+import _ from 'lodash';
+import {
+    ClinicalTrackDatum,
+    ClinicalTrackSpec,
+    IBaseHeatmapTrackDatum,
+    IBaseHeatmapTrackSpec,
+    IHeatmapTrackSpec,
+} from '../../../../shared/components/oncoprint/Oncoprint';
+import { makeUniqueColorGetter } from '../../../../shared/components/plots/PlotUtils';
+import { MUTATION_SPECTRUM_FILLS } from '../../../../shared/cache/ClinicalDataCache';
+import { MolecularProfile } from 'cbioportal-ts-api-client';
+import { AlterationTypeConstants } from '../../../resultsView/ResultsViewPageStore';
+import { capitalize } from 'cbioportal-frontend-commons';
+
+export const ONCOPRINTER_VAL_NA = 'N/A';
+
+export type OncoprinterOrderedValuesInputLine = {
+    sampleId: string;
+    orderedValues: string[];
+};
+
+type OncoprinterClinicalTrackSpec = {
+    trackName: string;
+    datatype: ClinicalTrackDataType;
+    countsCategories?: string[];
+};
+
+type OncoprinterHeatmapTrackSpec = {
+    trackName: string;
+    datatype: HeatmapTrackDataType;
+};
+
+type OncoprinterClinicalTrackDatum = Pick<
+    ClinicalTrackDatum,
+    'attr_id' | 'uid' | 'attr_val_counts' | 'na' | 'attr_val'
+> & { sample: string };
+
+type OncoprinterHeatmapTrackDatum = Pick<
+    IBaseHeatmapTrackDatum,
+    'profile_data' | 'uid' | 'na'
+> & { sample: string };
+
+// Output mimics mobxpromise form but thats just because its a nice way to package status and result.
+// This isn't meant to be plugged into mobxpromise machinery i.e. with `await`
+type ParseInputResult<TrackSpecType> =
+    | {
+          status: 'complete';
```

Comment:
'complete' implies to me some kind of async process and could lead to confusion given mobxpromise.    could we use 'parsedSucess' or 'success' or 'valid' ?

Reply:
good point, I will change that